src/braket/circuits/gates.py

```diff
@@ -39,16 +40,31 @@
     3. Register the class with the `Gate` class via `Gate.register_gate()`.
 """
 
+
+class _FiniteOrderGate(Gate, ABC):
```

Comment:
Missing docs. 

Why are these new superclasses needed?

Reply:
I like having these because many gates fall into this category. I think it's much better than pasting
```python
def adjoint_list(self):
    return [self]
```
for all involutory gates, and 
```python
def adjoint_list(self):
    return [self, self, self]  # or more, etc
```
for other orders. As well, it helps prevent stop a developer from missing or incorrectly implementing these methods for new gates in the same category.

Reply:
I suggest we remove both FiniteOrderGate and HermitianGate. It's probably more confusing for developers to use instead of just defining the adjoint operator themselves.

Reply:
I'm fine with that, as there aren't too many instances of these yet.